version: "3.9"

services:
  postgres:
    image: postgres:12.4-alpine
    volumes:
      - ./postgres_db/postgres_data:/var/lib/postgresql/data
    ports:
      - '5432:5432'
    environment:
      POSTGRES_USER: django
      POSTGRES_PASSWORD: django
      POSTGRES_DB: store
    restart: always
    healthcheck:
      test: ['CMD-SHELL', 'pg_isready  -U django -d store']
      interval: 5s
      timeout: 5s
      retries: 5
  migrations:
    build:
      context: .
    env_file:
      - ./.env
    depends_on:
      postgres:
        condition: service_healthy
    command: python manage.py migrate
  api:
    build:
      context: .
    ports:
      - '8000:8000'
    depends_on:
      postgres:
        condition: service_healthy
      migrations:
        condition: service_completed_successfully
    restart: always
    volumes:
      - ./.env:/app/.env

volumes:
  postgres_data:
